File: uk-ets-app-registry/uk-ets-registry-web/src/app/shared/model/transaction/signed-transaction-summary.ts
import {
  ReturnExcessAllocationTransactionSummary,
  TransactionSummary,
} from '@shared/model/transaction/transaction-summary';
import { SignatureInfo } from '@signing/model';

export interface SignedTransactionSummary extends TransactionSummary {
  signatureInfo: SignatureInfo;
}

export interface SignedReturnExcessAllocationTransactionSummary
  extends ReturnExcessAllocationTransactionSummary {
  signatureInfo: SignatureInfo;
}

File: uk-ets-app-registry/uk-ets-transaction-domain/src/main/java/gov/uk/ets/registry/api/transaction/domain/data/SignedTransactionSummary.java
package gov.uk.ets.registry.api.transaction.domain.data;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Represents a signed transaction summary.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SignedTransactionSummary extends TransactionSummary {

    private static final long serialVersionUID = -7592757131684871593L;

    private SignatureInfo signatureInfo;
}
